### PR TITLE
osbuild-tests: run compose tests in a temporary directory

### PR DESCRIPTION
**Pull request by larskarlitski**
Some `composer-cli` commands operate on the current directory, for
example saving blueprints or images. Add the `TemporaryWorkDir` type,
which helps changing to a temporary working directory and cleaning
everything up after.

The alternative would have been to pass a working directory to all calls
of `runComposerCLI`, but that seemed like it would require a lot of
change, which I didn't deem worth for testing code.

### CODE REVIEW

**Comment by larskarlitski** [comment]
Triggered by this PR: https://github.com/osbuild/osbuild-composer/pull/256

File: cmd/osbuild-tests/main.go [hunk 1]
```diff
@@ -71,13 +71,22 @@ func main() {
...
+	defer tmpdir.Close()
+
 	bp := blueprint.Blueprint{
 		Name:        "empty",
 		Description: "Test empty blueprint in toml format",
 	}
 	pushBlueprint(&bp)
 	defer deleteBlueprint(&bp)
 
+	runComposerCLI(false, "blueprints", "save", "empty")
```
Comment (teg) [inline cmd/osbuild-tests/main.go]: Hm, this hunk is unrelated to this patch, no?

Comment (larskarlitski) [reply cmd/osbuild-tests/main.go]: Indeed. I've split it into a separate commit.
